packages/cache/src/store.js

```diff
@@ -0,0 +1,55 @@
+// @ts-check
+import { Far } from '@endo/far';
+import { matches, makeScalarMapStore } from '@agoric/store';
+
+import '@agoric/store/exported.js';
+
+import { GROUND_STATE } from './state.js';
+
+const { details: X } = assert;
+
+/**
+ * @param {MapStore<unknown, import('./types').State>} [stateStore]
+ * @returns {import('./types').Coordinator}
+ */
+export const makeStoreCoordinator = (stateStore = makeScalarMapStore()) => {
+  const getCurrentState = key => {
+    if (!stateStore.has(key)) {
+      return GROUND_STATE;
+    }
+    return stateStore.get(key);
+  };
+  return Far('store cache coordinator', {
+    getRecentState: async key => getCurrentState(key),
```

Comment:
Are these `async`s required? Can this just be `getRecentState: getCurrentState,`?

Reply:
The signature calls for a promise return, so an async wrapper is the best way to ensure no synchronous throws.